src/ai/backend/manager/registry.py

```diff
@@ -693,12 +693,40 @@ async def enqueue_session(
             row  = await result.fetchone()
             dotfiles = msgpack.unpackb(row['dotfiles'])
             internal_data = {} if internal_data is None else internal_data
-            internal_data.update({'dotfiles': dotfiles})
             if row['ssh_public_key'] and row['ssh_private_key']:
                 internal_data['ssh_keypair'] = {
                     'public_key': row['ssh_public_key'],
                     'private_key': row['ssh_private_key'],
                 }
+            # use dotfiles in the priority of keypair > group > domain
+            dotfile_paths = set(map(lambda x: x['path'], dotfiles))
+            # add keypair dotfiles
+            internal_data.update({'dotfiles': list(dotfiles)})
+            # add group dotfiles
+            query = (sa.select([groups.c.dotfiles])
+                       .select_from(groups)
+                       .where(groups.c.id == group_id))
+            result = await conn.execute(query)
+            row = await result.fetchone()
+            dotfiles = msgpack.unpackb(row['dotfiles'])
```

Comment:
Seems like the exact same logic with the `query_group_dotfiles`. Why not use it?